tests/test_cog.py

```diff
@@ -43,13 +45,27 @@ def test_item_without_cogs(self) -> None:
         item.validate()
 
     def test_create_cogs(self) -> None:
-        infile = test_data.get_external_data(
-            "VNP09H1.A2012017.h00v09.001.2016294114238.h5"
-        )
-        with TemporaryDirectory() as temporary_directory:
-            paths = stactools.viirs.cog.cogify(infile, temporary_directory)
+        filename = "VNP09H1.A2012017.h00v09.001.2016294114238.h5"
+        href = test_data.get_external_data(filename)
+        _ = test_data.get_external_data(f"{filename}.xml")
+        with TemporaryDirectory() as tmp_dir:
+            paths = stactools.viirs.cog.cogify(href, tmp_dir)
             assert all(os.path.exists(path) for path in paths)
         self.assertEqual(len(paths), 5)
         file_names = [os.path.basename(path) for path in paths]
         expected_cog_names = COG_LIST
         self.assertEqual(set(file_names), set(expected_cog_names))
+
+    def test_int8toint16_nodata(self) -> None:
+        filename = "VNP13A1.A2022097.h11v05.001.2022113080900.h5"
+        href = test_data.get_external_data(filename)
+        _ = test_data.get_external_data(f"{filename}.xml")
+        with TemporaryDirectory() as tmp_dir:
+            paths = stactools.viirs.cog.cogify(href, tmp_dir)
+            for index, path in enumerate(paths):
+                if "pixel_reliability.tif" in path:
+                    break
+            native_int8_cog = paths[index]
```

Comment:
```suggestion
            native_int8_cog = next((path for path in paths if "pixel_reliability.tif" in path))
```

IMO this is cleaner, as it doesn't rely on the index variable escaping the `for` loop's scope.

Reply:
Yeah. Far better. Done.